packages/grid/_modules_/grid/models/rows.ts

```diff
@@ -44,7 +44,11 @@ export interface RowModel {
  * @param rowData Row as [[RowData]].
  * @returns A row as [[RowModel]].
  */
-export function createRow(rowData: RowData): RowModel {
+export function createRowModel(rowData: RowData): RowModel {
+  if (rowData.id == null) {
+    throw new Error(`Material-UI: Row without id found,  `);
```

Comment:
Maybe something in this order?

```suggestion
    throw new Error([
      'Material-UI: The data grid component requires all rows to have a unique id. ',
      'A row was provided without in the rows prop:',
      JSON.stringify(rowData),
   ]).join('\n');
```